### ISSUE TITLE
Normalize inputs to decoder

### ISSUE THREAD

**Issue by ajaychawda58** [marrrcin/rsrlayer-pytorch#1]
Hi, In the paper it says. "The output of the RSR layer is L2 normalized before applying the decoder".  

Therefore in the [code](https://github.com/marrrcin/rsrlayer-pytorch/blob/bc611cfdd4cb74398ca232aec90f68f96593f6e8/rsr_layer_pytorch.py#L73), The line should be 

`dec = self.decoder(F.normalize(latent, p=2))`.

Is it correct or you are normalizing it somewhere else?
